cgrok/grok_match.c

```diff
@@ -52,8 +52,7 @@ int grok_match_walk_next(const grok_match_t *gm,
   }
 
   *namelen = gct->name_len;
-  *name = malloc(*namelen);
-  memcpy(*name, gct->name, *namelen);
+  *name = gct->name;
```

Comment:
@alanctgardner  Were you able to find any more info why grok does it this way?

I'm +0 on this approach, but I know we need to get this fixed soon, so maybe we could add some strongly worded comments and leave it at that? I just don't want this to bite us later if we change the way we use the library and start using memory in undefined ways.